### DISCUSSION TITLE
Image exported to XLSX not centered. Label contents is not wrapped when exported to XLSX

### DISCUSSION THREAD

**Discussion by raviprsd18** [eclipse-birt/birt#1546] [Q&A] [answered]
Hello All,

Following are my issues. Any help will be appreciated.

1. My report design have a image, which is placed inside the table => row => cell. The cell property is set to center as a result the image is displayed in center in birt viewer and pdf file. But it is not working when exported to xlsx file type.

2. The label contents are wrapped both birt viewer and pdf file but not in xlsx.

**BIRT Viewer**
![image](https://github.com/eclipse-birt/birt/assets/46317080/f75cd4d1-ae8a-4ffb-bc65-93a086423fdb)

PDF **Viewer**
![image](https://github.com/eclipse-birt/birt/assets/46317080/79ec1c4d-d41f-44ab-987c-d4b7a89662e7)

**XLSX File**
![image](https://github.com/eclipse-birt/birt/assets/46317080/7f37e6cb-b8d8-4387-9740-bc34525a1313)

**Comment by speckyspooky** [comment]
The topic is that the image will be handled in anotherway in excel instead of HTML, PDF, WORD or PPT.
In excel isn't the image an integrated part of the table cell. The other mentioned file types handle the images as cell content.
In excel is a image only an overlay object. This behavior is given on excel for each drawing element also of charts for example.
Therefore the style "center" is not working.

I took a look into the API of excel and there a technical option to manipulate the image position which is called anchor-mechanism
but this way isn't implemented yet and must be implemented for all according orientations of the cell.

A workaround could be that you create 3 columns into your grid. The middle column will contains your icon and with the left & right column you will get the final width of your table. For your text you can merge the row-cells to 1 big cell.

I use this workaround for my excel eports were I have company logos at special positions.

**Reply by raviprsd18** [reply]
Thanks @speckyspooky for the answer. I will definitely try.

Would you please help me in the point 2 issue also ?

**Comment by speckyspooky** [comment, marked as answer]
@raviprsd18 

Attached you will find a small example of your report with correct wrapped text.
For that I added a small script on the text element which changes the wrap option of the text - onRender().
The style wrap property will be changed if excel file will be created - I hope this will help you.

[icon_central.zip](https://github.com/eclipse-birt/birt/files/14014844/icon_central.zip)

**Reply by hvbtup** [reply]
Yes, I vaguely remember that this could be solved by explicitly setting the white-space property. 
Note that the example report which Thomas attached contains some other scripts which are not related to the problem.

**Comment by raviprsd18** [comment]
Thanks @speckyspooky. All the solutions provided are working as per my needs.
